functions/utilities.zsh

```diff
@@ -111,6 +111,15 @@ case $(uname) in
       ;;
 esac
 
+# Not all OSes support the '-o' parameter
+# That's why this second condition is needed 
+case $(uname -o 2>/dev/null) in
```

Comment:
Hm, I see why you did this, but at the same time, breaking this out of the main `case` statement will result in every OS evaluating this conditional, even if the OS has already been discovered.

What does `uname` return on Android? `Linux`? If so, perhaps we can add this as a second conditional within the `Linux` case in the main statement.